Makefile

```diff
@@ -1,17 +1,44 @@
-BINARY_NAME=fga
-BUILD_FOLDER=dist
+#-----------------------------------------------------------------------------------------------------------------------
+# Variables (https://www.gnu.org/software/make/manual/html_node/Using-Variables.html#Using-Variables)
+#-----------------------------------------------------------------------------------------------------------------------
+BINARY_NAME = fga
+BUILD_DIR ?= $(CURDIR)/dist
+GO_BIN ?= $(shell go env GOPATH)/bin
 
-all: build
+#-----------------------------------------------------------------------------------------------------------------------
+# Rules (https://www.gnu.org/software/make/manual/html_node/Rule-Introduction.html#Rule-Introduction)
+#-----------------------------------------------------------------------------------------------------------------------
+$(GO_BIN)/golangci-lint:
+	@echo "==> Installing golangci-lint within "${GO_BIN}""
+	@go install -v github.com/golangci/golangci-lint/cmd/golangci-lint@latest
 
-build:
-	go build -o ${BUILD_FOLDER}/${BINARY_NAME} main.go
+$(GO_BIN)/govulncheck:
+	@echo "==> Installing govulncheck within "${GO_BIN}""
+	@go install -v golang.org/x/vuln/cmd/govulncheck@latest
+
+$(GO_BIN)/gofumpt:
+	@echo "==> Installing gofumpt within "${GO_BIN}""
+	@go install -v mvdan.cc/gofumpt@latest
+
+$(GO_BIN)/gci:
+	@echo "==> Installing gci within "${GO_BIN}""
+	@go install -v github.com/daixiang0/gci@latest
+
+$(BUILD_DIR)/$(BINARY_NAME):
+	@echo "==> Building binary within ${BUILD_DIR}/${BINARY_NAME}"
+	go build -v -o ${BUILD_DIR}/${BINARY_NAME} main.go
+
+.PHONY: build run clean test lint audit format
+
+build: $(BUILD_DIR)/$(BINARY_NAME)
  
-run: build
-	./${BUILD_FOLDER}/${BINARY_NAME}
+run: $(BUILD_DIR)/$(BINARY_NAME)
```

Comment:
This wasn't working as intended before. By allowing $(ARGS) we can now actually make use of this and pass whatever arguments and flags we want to the underlying `fga` binary.

For example: `make run $ARGS="tuple write -h"`